components/MakeFunction.coffee

```diff
@@ -1,61 +1,52 @@
 noflo = require 'noflo'
 
-class MakeFunction extends noflo.Component
-  description: 'Evaluates a function each time data hits the "in" port
+exports.getComponent = ->
+  c = new noflo.Component
+  c.description = 'Evaluates a function each time data hits the "in" port
   and sends the return value to "out". Within the function "x" will
   be the variable from the in port. For example, to make a ^2 function
   input "return x*x;" to the function port.'
-  icon: 'code'
-
-  constructor: ->
-    @f = null
-
-    # We have two input ports. One for the callback to call, and one
-    # for IPs to call it with
-    @inPorts = new noflo.InPorts
-      in:
-        datatype: 'all'
-        description: 'Packet to be processed'
-      function:
-        datatype: 'string'
-        description: 'Function to evaluate'
-    # The optional error port is used in case of wrong setups
-    @outPorts = new noflo.OutPorts
-      out:
-        datatype: 'all'
-      function:
-        datatype: 'function'
-      error:
-        datatype: 'object'
-
-    # Set callback
-    @inPorts.function.on 'data', (data) =>
-      if typeof data is "function"
-        @f = data
-      else
+  c.icon = 'code'
+
+  c.inPorts.add 'in',
+    datatype: 'all'
+    description: 'Packet to be processed'
+  c.inPorts.add 'function',
+    datatype: 'string'
```

Comment:
It kind of does. But the primary input is expected to be a string to evaluate. It is just that NoFlo doesn't actually enforce types, so we could just as well accept a real function